go/database/mpt/hasher.go

```diff
@@ -458,9 +458,11 @@ func (h ethHasher) getHash(ref *NodeReference, source NodeSource) (common.Hash,
 		return common.Hash{}, err
 	}
 
-	// The hash for embedded nodes is 0.
+	// The hash for embedded nodes is the node representation.
 	if len(data) < 32 {
-		return common.Hash{}, nil
+		var hash common.Hash
```

Comment:
Please add a unit test in `hasher_test.go` for this to document this requirement.

Reply:
added, see` TestEthereumLikeHasher_EmbeddedNode_Hashes_As_Itself`